### DISCUSSION TITLE
leafcli not installing

### DISCUSSION THREAD

**Discussion by vimkaf** [leafsphp/leaf#143] [Q&A] [answered]
Good day house, I am having some composer troubles when installing leafcli

Command
`composer global require leafs/cli `

Error
 Problem 1
    - Root composer.json requires leafs/cli ^2.4 -> satisfiable by leafs/cli[v2.4.0, v2.4.1, v2.4.2].
    - leafs/cli[v2.4.0, ..., v2.4.2] require psy/psysh ^0.11.1 -> found psy/psysh[v0.11.1, ..., v0.11.8] but the package is fixed to v0.10.12 (lock file version) by a partial update and that version does not match. Make sure you list it as an argument for the update command.

Use the option --with-all-dependencies (-W) to allow upgrades, downgrades and removals for packages currently locked to specific versions. You can also try re-running composer require with an explicit version constraint, e.g. "composer require leafs/cli:*" to figure out if any version is installable, or "composer require leafs/cli:^2.1" if you know which you need.

**Comment by mychidarko** [comment]
Please try `composer global require leafs/cli --with-all-dependencies`

**Reply by vimkaf** [reply]
i'm still getting errors

 Problem 1
    - Root composer.json requires leafs/cli ^2.4 -> satisfiable by leafs/cli[v2.4.0, v2.4.1, v2.4.2].
    - leafs/cli[v2.4.0, ..., v2.4.2] require psy/psysh ^0.11.1 -> found psy/psysh[v0.11.1, ..., v0.11.8] but these were not loaded, likely because it conflicts with another require.

You can also try re-running composer require with an explicit version constraint, e.g. "composer require leafs/cli:*" to figure out if any version is installable, or "composer require leafs/cli:^2.1" if you know which you need.

Installation failed, reverting ./composer.json and ./composer.lock to their original content.

**Comment by mychidarko** [comment, marked as answer]
This as well? `composer global require leafs/cli:*`

**Reply by vimkaf** [reply]
This worked

**Reply by mychidarko** [reply]
Can you type out `leaf --version` to check which version was installed?

**Reply by vimkaf** [reply]
cannot find leaf command

**Reply by mychidarko** [reply]
You can follow the instructions on this page to add it to your global path

https://cli.leafphp.dev/docs/#installation

**Reply by mychidarko** [reply]
That way you'll have access to the `leaf` command

**Reply by vimkaf** [reply]
The composer vendor bin dir is already in my path

**Reply by mychidarko** [reply]
That's weird then

**Reply by mychidarko** [reply]
Did you by chance grab the version installed from this command? `composer global require leafs/cli:*`

**Reply by vimkaf** [reply]
I noticed that it installed it in the current working directory so i cd into ~ and ran `composer global require leafs/cli:*` now i'm getting this error below

![image](https://user-images.githubusercontent.com/2923113/189117352-5d4cafa7-9497-40c5-8399-b64df9d33a1f.png)

**Reply by mychidarko** [reply]
Which PHP version are you currently running?

**Reply by vimkaf** [reply]
PHP 8.1.9

**Reply by mychidarko** [reply]
It's attempting to install way older versions. I'll update the CLI package and see if this helps with your issue. You can open an issue on https://github.com/leafsphp/cli

**Reply by mychidarko** [reply]
Thanks @vimkaf

**Reply by vimkaf** [reply]
Glad to help @mychidarko Could you let me know when you have updated it so i could try it again.

**Reply by mychidarko** [reply]
Sure, will do that

**Reply by mychidarko** [reply]
@vimkaf pls try again now

**Reply by vimkaf** [reply]
which of the command should i run?

**Reply by mychidarko** [reply]
`composer global require leafs/cli`

**Reply by vimkaf** [reply]
It's working now :)
![image](https://user-images.githubusercontent.com/2923113/189138010-3bd48f43-8e62-4553-b91d-58d81260f4ae.png)

**Reply by vimkaf** [reply]
Thanks for your time @mychidarko

**Reply by mychidarko** [reply]
Nice 🔥
